plugins/catalog-backend/src/ingestion/processors/PlaceholderProcessor.ts

```diff
@@ -103,7 +103,7 @@ export class PlaceholderProcessor implements CatalogProcessor {
       const resolverKey = keys[0].substr(1);
       const resolverValue = data[keys[0]];
       const resolver = this.options.resolvers[resolverKey];
-      if (!resolver || typeof resolverValue !== 'string') {
+      if (!resolver) {
         // If there was no such placeholder resolver or if the value was not a
```

Comment:
Given the context in this comment I'm not sure this is a good idea tbh. Did you explore the option of creating a custom processor that handles this instead, rather than taking the route of a custom placeholder resolver? With a custom processor you get a lot more freedom and can make a bit more of a first-class citizen of your catalog schema. I'm thinking things like adding a processor that for example is able to read this:

```yaml
dependsOn:
  - 'Component:<actual relation>'
dependencyIntegrations:
  datadog:
    name: '<datadog-service-name>'
    type: 'calls'
```

You'd be emitting those relations directly in the processor instead, rather than taking the route through the `dependsOn` field. You can see an example of how that's done here: https://github.com/backstage/backstage/blob/81c2cee536281ed65a7ee06f7e2491bdff3bf31b/plugins/scaffolder-backend/src/processor/ScaffolderEntitiesProcessor.ts#L72-L93

Reply:
I couldn't find a way to just add on an extra step to the processing. For my use case, this was needing to apply to a custom entity provider, a custom processor, and the default one.

Is there a way to hook into a step past all of these so i'm not having to do away with the default processor, and manually call this bit of functionality in three places? Like some kind of "post processor", or some hook that happens after the initial processing?

Reply:
@Rugvip I could be missing or miss understanding an api though. Is this kind of "processing pipeline" something that one can tap into or is being considered?

Reply:
Hmmm. I could have sworn I tried that first and it wasn't picking up the values from the entity provider. I'll give it another try and see.

Reply:
@Rugvip ended up using a `preProcessEntity` on a `CatalogProcessor` and it works great. Thanks for the tip. I don't know what I was doing before that caused it not to be seen. 🤔 

Anyway, this works much better for this usecase. If I come back across a valid use case for a JSON placeholder, I will reopen this PR. Thanks!